### PR TITLE
Check data dimensions shape

### PR DESCRIPTION
**Pull request by jdahm**
## Description

Adds a check for data dimensions shape and a test.

### CODE REVIEW

File: src/gt4py/stencil_object.py [hunk 1]
```diff
@@ -267,6 +267,15 @@ def _validate_args(self, field_args, param_args, domain, origin) -> None:
                         f"expects {field_domain_ndim + len(field_info.data_dims)} ('{field_info.axes}[{field_info.data_dims}]')"
                     )
 
+                # Check: data dimensions shape
+                if not all(
+                    field.shape[field_domain_ndim + i] == dim_size
+                    for i, dim_size in enumerate(field_info.data_dims)
+                ):
```
Comment (egparedes) [inline src/gt4py/stencil_object.py:274]: Maybe this check can be simplified like this:
```suggestion
                if field.shape[field_domain_ndim:] != field_info.data_dims:
```
Or am I missing something? Can `field.shape` have more dimensions?

**Comment by egparedes** [comment]
bors try

**Review by egparedes — COMMENTED** [review]
It looks good to me, just a single comment

File: src/gt4py/stencil_object.py (hunk 1 above)
Comment (jdahm) [reply src/gt4py/stencil_object.py:274]: No it cannot as far as I know. This is a nice simplification!

**Pull request merged**